net/tshttpproxy/tshttpproxy_synology.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build linux
+// +build linux
+
+package tshttpproxy
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+
+	"tailscale.com/util/lineread"
+	"tailscale.com/version/distro"
+)
+
+func init() {
+	if distro.Get() == distro.Synology {
+		sysProxyFromEnv = synologyProxyFromConfigCached
+	}
+}
+
+var cachedProxy struct {
+	sync.Mutex
+	val        *url.URL
+	lastUpdate time.Time
+}
+
+func synologyProxyFromConfigCached(req *http.Request) (*url.URL, error) {
+	cachedProxy.Lock()
+	defer cachedProxy.Unlock()
+
+	if time.Since(cachedProxy.lastUpdate) < time.Minute {
```

Comment:
time.Since isn't super well defined when working on a zero time.Time: https://go.dev/play/p/P7S_BY2iHJe

It should be considered random.

Instead: `if time.Now().Before(cachedProxy.lastUpdate.Add(time.Minute))` or something, perhaps pulling part of that out into a `tooOld` or `expiry` variable.

Reply:
The zero value of time.Time has a documented well defined wall time, and the output seems to agree with that.

I assume the source of potential instability comes from the monotonic value, which is signed. Is there some condition in which the monotonic value on the system reads as a sub-0 time?

Reply:
I used IsZero, as I started reading through the behaviors in time post monotonic patches, and it's a bit of work I'd rather just avoid needing to think about for the Add case as well, which appears to also have monotonic portability / range concerns.